pool/watcher.go

```diff
@@ -75,7 +75,7 @@ type poolWatcher struct {
 	mode     Mode
 	key      string
 	callback tarantool.WatchCallback
-	// watchers is a map connection -> connection watcher.
+	// watchers is a map connection id -> connection watcher.
 	watchers map[string]tarantool.Watcher
```

Comment:
Why do we need the id here at all?
```suggestion
	watchers map[*tarantool.Connection]tarantool.Watcher
```